### ISSUE TITLE
UTF-8 in URL in source generates truncated URL in HTML

### ISSUE THREAD

**Issue by paul-j-lucas** [doxygen/doxygen#8241]
**Describe the bug**
A Doxygen comment in C code contains:

    @sa https://en.wikipedia.org/wiki/Damerau–Levenshtein_distance

Note that the "–" there is a Unicode En Dash (U+2013) and is in the `.c` file encoded as UTF-8: E2 80 93.

The generated HTML is:

    <a href="https://en.wikipedia.org/wiki/Damerau">https://en.wikipedia.org/wiki/Damerau</a>–Levenshtein_distance

That is, the URL gets truncated at the en-dash character and the remaining part of the URL is put after the `</a>`.  Of course this means that the URL is wrong.

If you're going to suggest using percent-encoded characters, i.e.:

    https://en.wikipedia.org/wiki/Damerau%E2%80%93Levenshtein_distance

that doesn't work either.  In that case, the generated HTML is:

    <a href="https://en.wikipedia.org/wiki/Damerau%E2%80%93Levenshtein_distance">https://en.wikipedia.org/wiki/Damerau%E2%80%93Levenshtein_distance</a>

i.e., the percent-encoded characters are preserved as-is rather than interpreted as being in UTF-8 and they're displayed as such in a browser which is wrong. (However, the link displayed in the URL tooltip at the bottom of a Safari window _is_ correct since Safari does the correct interpretation of the UTF-8. Hence, the correct link is clickable. It just _looks_ wrong on the screen.

**Expected behavior**
I expect the HTML generated _should_ be:

    <a href="https://en.wikipedia.org/wiki/Damerau–Levenshtein_distance">https://en.wikipedia.org/wiki/Damerau–Levenshtein_distance</a>

That is, I expect your URL parser _not_ to stop at the start of a UTF-8 encoded character. Your parser should stop only upon encountering one of a whitespace, `<`, or `>` character.

**To Reproduce**
```
/**
 * Implements the Damerau-Levenshtein algorithm to calculate an _edit distance_
 * between strings, i.e., how many letters need to be swapped within,
 * substituted within, deleted from, or added to \a source to get \a target.
 *
 * @param source The source string.
 * @param target The target string.
 * @return Returns said distance.
 *
 * @sa https://en.wikipedia.org/wiki/Damerau–Levenshtein_distance
 * @sa https://www.lemoda.net/text-fuzzy/damerau-levenshtein/
 */
dam_lev_t dam_lev_dist( char const *source, char const *target );
```

**Version**
On macOS Catalina 10.15.7
```
$ doxygen --version
1.8.20
```

**Comment by albert-github** [comment]
Fortunately there is a workaround in doxygen by using markdown type of links:
```
 * @sa [https://en.wikipedia.org/wiki/Damerau–Levenshtein_distance](https://en.wikipedia.org/wiki/Damerau–Levenshtein_distance)
 * @sa [https://www.lemoda.net/text-fuzzy/damerau-levenshtein/](https://www.lemoda.net/text-fuzzy/damerau-levenshtein/)
```
which results in a correct link.

(Also inputting by hand the `–` is probably a problem, on wikipedia I see that the normal dash / minus sign `-` works as well and does a redirect).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8243

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
